dirigera-client-mqtt/src/main/java/de/dvdgeisler/iot/dirigera/client/mqtt/DirigeraClientMqttApplication.java

```diff
@@ -52,13 +53,20 @@ public MqttClient getMqttClient(@Value("${dirigera.mqtt.hostname:localhost}") fi
         final String uri;
 
         publisherId = api.status().map(s -> s.id).block();
-        uri = String.format("tcp://%s:%d", host, port);
+        options = new MqttConnectOptions();
+
+        if (port == 8883) {
```

Comment:
I think it would make more sense to introduce a command line parameter (e.g. `ssl=true`) than go of a specific port.

This makes it a lot more flexible for users and less error prone

Reply:
Would it make sense to let the whole URL configurable instead? And, provide a way to reconfigure the default SSLSocketFactory for custom CA needs?

Reply:
In terms of the SSLSocketFactory with custom CA needs, I would say lets implement a MVP first (this PR) and if more customisation is needed in the future, we can adjust it.

The URL part is already kind configurable. I'm not sure if we want to provide a "free" format for it at this point in time.
Again this is up for debate and potential future updates. In my opinion focus on the MVP to get an initial version out and see if we've additional features to develop :)

Reply:
Added. A new parameter: `dirigera.mqtt.use-ssl`